packages/core/src/lib/core/mutations/hooks.ts

```diff
@@ -1,43 +1,28 @@
 import { extensionError } from '../graphql-errors';
+import type { InitialisedList } from '../initialise-lists';
 
 export async function runSideEffectOnlyHook<
-  HookName extends string,
-  List extends {
-    fields: Record<
-      string,
-      {
-        hooks: {
-          [Key in HookName]?: (args: { fieldKey: string } & Args) => Promise<void> | void;
-        };
-      }
-    >;
-    hooks: {
-      [Key in HookName]?: (args: any) => Promise<void> | void;
-    };
-    listKey: string;
-  },
-  Args extends Parameters<NonNullable<List['hooks'][HookName]>>[0]
->(list: List, hookName: HookName, args: Args) {
-  // Runs the before/after operation hooks
-
+  HookName extends 'beforeOperation' | 'afterOperation',
+  Args extends Parameters<NonNullable<InitialisedList['hooks'][HookName]>>[0]
+>(list: InitialisedList, hookName: HookName, args: Args) {
   let shouldRunFieldLevelHook: (fieldKey: string) => boolean;
   if (args.operation === 'delete') {
-    // Always run field hooks for delete operations
+    // always run field hooks for delete operations
     shouldRunFieldLevelHook = () => true;
   } else {
-    // Only run field hooks on if the field was specified in the
-    // original input for create and update operations.
+    // only run field hooks on if the field was specified in the
+    //   original input for create and update operations.
     const inputDataKeys = new Set(Object.keys(args.inputData));
     shouldRunFieldLevelHook = fieldKey => inputDataKeys.has(fieldKey);
   }
 
-  // Field hooks
+  // field hooks
   const fieldsErrors: { error: Error; tag: string }[] = [];
   await Promise.all(
     Object.entries(list.fields).map(async ([fieldKey, field]) => {
       if (shouldRunFieldLevelHook(fieldKey)) {
         try {
-          await field.hooks[hookName]?.({ fieldKey, ...args });
+          await field.hooks[hookName]({ fieldKey, ...args } as any); // TODO: FIXME any
```

Comment:
I think `runSideEffectOnlyHook` will be removed soon, so this is less of a concern